docs/core/development/policy/coding-standard.rst

```diff
@@ -625,6 +625,126 @@ Any attribute, method argument, or method name that corresponds *directly* to a
 All SQL keywords should be in upper case.
 
 
+Emitting logs
+-------------
+
+All new code emitting log messages should use the new logging system.
+
+This section is dedicated to using the new logging system in Twisted core code.
+For general information about using the logging system please check the dedicated documentation about :doc:`how to log with twisted.logger </core/howto/logger>` and the :api:`twisted.logger.Logger <Logger>` API docs.
+
+General usage rules:
+
+* Don't use explicit namespaces.
+  The logger will automatically fill the namespace.
+* Use the global log publisher.
+  Twisted code may call into non-Twisted code which makes stdlib/legacy log messages, which then get redirected to the global logger.
+* When possible instantiate the Logger as ``self._log`` instance member.
+* Emitted logs are not part of the API/contract so you don't have to document it as part of the API.
+* For security features you might documented the emitted logs and then they will became part of the contract.
+
+Example of usage:
+
+.. code-block:: python
+
+    from twisted.logger import Logger
+
+    class SomeStuff(object):
+        """
+        Class using the new logging system as an instance member
+        """
+
+        _log = Logger()
+
+        def getUser(self, name):
+            try:
+                return self._otherStuff()
+            except Exception as error:
+                self._log.info(
+                    "Unable to check {user!s}", user=name, error=error)
+
+                raise UnauthorizedLogin()
+
+
+General testing rules:
+
+* New code calling the new logging system will test that the log is emitted.
```

Comment:
This rule also doesn't motivate or explain itself, so it comes across as somewhat arbitrary.  Why does it need to test that the log is emitted?

"As with any behavior, if it's worth implementing, it's worth testing.  Changes which introduce or modify a log message should test the relevant aspects of the log message's behavior."